realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -112,6 +112,7 @@ public boolean handleMessage(Message message) {
                 }
                 case COMPLETED_UPDATE_ASYNC_QUERIES: {
                     // this is called once the background thread completed the update of the async queries
+                    notifyAllListeners();
```

Comment:
I _think_ it is OK since REALM_CHANGED will only be triggered if the caller thread has advanced since it started its async query, but I hope @nhachicha will review this throughly.